### PR TITLE
Regression fix sp arth and refs

### PR DESCRIPTION
**Pull request by JodiTheTigger**
Three c++ 0.73 regression fixes here regarding `&`

### CODE REVIEW

File: src/combine.cpp [hunk 1]
```diff
@@ -1538,12 +1538,16 @@ void do_symbol_check(Chunk *prev, Chunk *pc, Chunk *next)
...
+      Chunk *prevNext = prev->GetNext();
+
       if (prev->Is(CT_DELETE))
       {
          pc->SetType(CT_ADDR);
       }
       else if (  prev->Is(CT_TYPE)
-              || prev->Is(CT_QUALIFIER))
+              || prev->Is(CT_QUALIFIER)
+              || (  prevNext
```
Comment (micheleCTDE) [inline src/combine.cpp:1549]: No need to test for "prev" to be not null. "GetNext()" is degined to never return a nullptr.

**Review by micheleCTDE — CHANGES_REQUESTED** [review]
Please amend as per comments

File: src/combine.cpp [hunk 2]
```diff
@@ -1616,7 +1621,23 @@ void do_symbol_check(Chunk *prev, Chunk *pc, Chunk *next)
...
                   else if (tmp->Is(CT_DC_MEMBER))
                   {
-                     prev->SetType(CT_TYPE);
+                     // Issue #2103 & Issue #3865: partial fix
+                     // No easy way to tell between an enum and a type with
+                     // a namespace qualifier. Compromise: if we're in a
+                     // function def or call, assume it's a ref.
+                     Chunk *nextNext = next->GetNext();
+
+                     if (  (!nextNext || nextNext->IsNot(CT_DC_MEMBER))
```
[the remark this answers is not in the document]
Comment (JodiTheTigger) [reply src/combine.cpp:1630]: thanks for the comments, will do when I get to my computer next.

File: src/combine.cpp (hunk 1 above)
Comment (JodiTheTigger) [reply src/combine.cpp]: done.

File: src/combine.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (JodiTheTigger) [reply src/combine.cpp]: done.

**Review by micheleCTDEAdmin — APPROVED** [review]
Looks good

**Review by micheleCTDE — APPROVED** [review]
